### PR TITLE
🌐 Add Chinese translation for `docs/zh/docs/tutorial/dependencies/dependencies-with-yield.md`

### PR DESCRIPTION
**Pull request by jaystone776**
…ield.md

as title

### CODE REVIEW

File: docs/zh/docs/tutorial/dependencies/dependencies-with-yield.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+    生成器依赖项可以使用**异步**函数或普通函数。
+    
+    与普通依赖项一样，**FastAPI** 会正确处理每个生成器依赖项。
+
+## 在依赖项中使用 `yield` 和 `try`
+
+把依赖项的 `yield` 置于 `try`  代码块中，就可以接收到所有使用该依赖项时触发的异常。
+
+假设执行过程中的某行代码（这行代码可能在*路径操作或其他依赖项中*）导致了数据库事务的「回滚」，或导致了任何其他错误，使用 `try`  就可以在当前依赖项中触发异常。
```
Comment (waynerv) [inline docs/zh/docs/tutorial/dependencies/dependencies-with-yield.md:62]:
```suggestion
假设执行过程中的某行代码（这行代码可能在*路径操作或其他依赖项中*）导致了数据库事务的「回滚」，或导致了任何其他错误，使用 `try` 就可以在当前依赖项中触发异常。
```

File: docs/zh/docs/tutorial/dependencies/dependencies-with-yield.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+您可能会想在依赖项的退出代码中，即 `yield` 之后，触发 `HTTPException` 或类似的异常，但实际上这种方式**行不通**。
+
+生成器依赖项的退出代码是在[异常处理器](../handling-errors.md#install-custom-exception-handlers)之后执行的。也就是说，在依赖项退出代码（`yield` 语句之后）中触发的异常不会被捕获。
+
+因此，如果在 yield 语句后触发 HTTPException，本应捕获 HTTPException 并返回 HTTP 400 响应的默认（或自定义）异常处理器将不会工作。
+
+通过这种方式，**FastAPI** 允许依赖项中的任意对象（例如，数据库会话）都可以用于后台任务。
+
+后台任务是在响应发送*后*执行的任务。所以，此时抛出  HTTPException  没有意义，因为没有办法改变*已发送*的响应。
```
Comment (waynerv) [inline docs/zh/docs/tutorial/dependencies/dependencies-with-yield.md:120]:
```suggestion
后台任务是在响应发送*后*执行的任务。所以，此时抛出 HTTPException 没有意义，因为没有办法改变*已发送*的响应。
```

File: docs/zh/docs/tutorial/dependencies/dependencies-with-yield.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+但是，如果后台任务导致了数据库错误，至少可以在生成器依赖项中执行回滚操作，或干净地关闭会话，而且还可以生成错误日志，或向远程跟踪系统发送错误报告。
+
+如果预计某些代码有可能触发异常，最「Pythonic」的做法就是在这段代码中添加 `try` 异常处理代码块。
+
+如果需要在*返回响应*前处理自定义的异常，并且需要修改响应内容，或要触发 HTTPException，可以创建[自定义异常处理器](../handling-errors.md#install-custom-exception-handlers)。
+
+!!! tip "提示"
+
+    不要在 `yield` 语句之后，而是要在该语句之前触发 `HTTPException` 等异常。
```
Comment (waynerv) [inline docs/zh/docs/tutorial/dependencies/dependencies-with-yield.md:130]:
```suggestion
    在依赖项的 `yield` 语句之前抛出 `HTTPException` 等异常仍然是可行的，但不能在 `yield` 之后。
```

**Review by waynerv — APPROVED** [review]
Excellent works!

**Comment by tiangolo** [comment]
📝 Docs preview for commit 979ade870b9b1272d65bb19697869a291db22e6a at: https://6499d1a2bac8cd0a1954b002--fastapi.netlify.app

**Comment by alejsdev** [comment]
Hi! This one was already handled in https://github.com/tiangolo/fastapi/pull/10870. For now, I'll close this one, but thanks for the help @jaystone776 @waynerv 🚀

**Pull request closed without merging**